pulse/data-sync-with-bigquery/index.ts

```diff
@@ -0,0 +1,89 @@
+import 'dotenv/config'
+import { PrismaClient, User } from '@prisma/client'
+import { withPulse } from '@prisma/extension-pulse'
+import { BigQuery } from '@google-cloud/bigquery'
+
+const DATASET_ID = 'pulse_demo'
+const TABLE_ID = 'users'
+
+const bigquery = new BigQuery()
+
+const apiKey: string = process.env.PULSE_API_KEY ?? ''
+const prisma = new PrismaClient().$extends(withPulse({ apiKey: apiKey }))
+
+async function main() {
+
+  // Setup
+  await createDataset()
+  await createTable()
+
+  // Start streaming database change events with Pulse
+  const stream = await prisma.user.stream({
+    create: {},
+    name: 'all-create-events'
+  })
+
+  process.on('exit', (code) => {
+    stream.stop()
+  })
+
+  for await (const event of stream) {
+    console.log('Just received an event:', event)
+    const user: User = event.created
+    syncData(user)
+  }
+}
+
+
+
+async function syncData(user: User) {
```

Comment:
I'd avoid storing the email from the user object. Due to compliance [issues](https://medium.com/@florian.trehaut/ensuring-gdpr-compliance-on-gcp-bigquery-efficiently-managing-the-right-to-be-forgotten-a76137944633) popping regarding storing PIIs such as email.